src/Core/Services/GraphQLSchemaCreator.cs

```diff
@@ -251,6 +270,124 @@ private DocumentNode GenerateSqlGraphQLObjects(RuntimeEntities entities, Diction
+                    string linkingEntityName = RuntimeConfig.GenerateLinkingEntityName(sourceEntityName, targetEntityName);
+                    ObjectTypeDefinitionNode targetNode = objectTypes[targetEntityName];
+                    IEnumerable<ForeignKeyDefinition> foreignKeyDefinitions = sourceDbo.SourceDefinition.SourceEntityRelationshipMap[sourceEntityName].TargetEntityToFkDefinitionMap[targetEntityName];
+
+                    // Get list of all referencing columns in the linking entity.
+                    List<string> referencingColumnNames = foreignKeyDefinitions.SelectMany(foreignKeyDefinition => foreignKeyDefinition.ReferencingColumns).ToList();
+
+                    NameNode sourceTargetLinkingNodeName = new(LINKING_OBJECT_PREFIX + objectTypes[sourceEntityName].Name.Value + objectTypes[targetEntityName].Name.Value);
+                    List<FieldDefinitionNode> fieldsInSourceTargetLinkingNode = targetNode.Fields.ToList();
+                    List<FieldDefinitionNode> fieldsInLinkingNode = linkingObjectTypes[linkingEntityName].Fields.ToList();
+
+                    // Store the names of relationship/column fields in the target entity to prevent conflicting names
+                    // with the linking table's column fields.
+                    HashSet<string> fieldNamesInTarget = targetNode.Fields.Select(field => field.Name.Value).ToHashSet();
+
+                    // The sourceTargetLinkingNode will contain:
+                    // 1. All the fields from the target node to perform insertion on the target entity,
+                    // 2. Fields from the linking node which are not a foreign key reference to source or target node. This is required to get values for
+                    // all the columns in the linking entity other than FK references so that insertion can be performed on the linking entity.
+                    foreach (FieldDefinitionNode fieldInLinkingNode in fieldsInLinkingNode)
+                    {
+                        string fieldName = fieldInLinkingNode.Name.Value;
+                        if (!referencingColumnNames.Contains(fieldName))
+                        {
+                            if (fieldNamesInTarget.Contains(fieldName))
+                            {
+                                // The fieldName can represent a column in the targetEntity or a relationship.
+                                // The fieldName in the linking node cannot conflict with any of the
+                                // existing field names (either column name or relationship name) in the target node.
+                                bool doesFieldRepresentAColumn = sqlMetadataProvider.TryGetBackingColumn(targetEntityName, fieldName, out string? _);
+                                string infoMsg = $"Cannot use field name '{fieldName}' as it conflicts with one of the other field's name in the entity: {targetEntityName}. ";
+                                string actionableMsg = doesFieldRepresentAColumn ?
```

Comment:
will this run as part of `dab validate` at some point?

Reply:
Yep. We can make this a part of dab validate later. I'll open an issue to track this.

Reply:
https://github.com/Azure/data-api-builder/issues/2031